packages/toolpad-utils/src/collections.spec.ts

```diff
@@ -11,3 +11,45 @@ describe('asArray', () => {
     expect(asArray(got)).toEqual(expected);
   });
 });
+describe('isReferEqual', () => {
+  test('isReferEqual array', () => {
+    const m = [1, 2, 3];
+    const d = [1, 2, 4];
+    expect(isReferEqual(m, m)).toBe(true);
```

Comment:
This is the same variable `m` which I think means both values will be referentially equal?
So it might be a better test to pass a different instance of the same array as each argument -  you can pass it directly like `isReferEqual([1, 2, 3], [1, 2, 3])`, for example.
Same for similar cases below.